From 6c2436c81bdc9811e4df9e27acd82994480acda7 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Sat, 30 May 2026 21:12:07 -0400
Subject: [PATCH] winewayland: Require wl_seat version 3.

We use wl_pointer_release unconditionally, which means we must bind to version 3 or higher.
---
 dlls/winewayland.drv/wayland.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/winewayland.drv/wayland.c b/dlls/winewayland.drv/wayland.c
index d884794c01a..56297dedc81 100644
--- a/dlls/winewayland.drv/wayland.c
+++ b/dlls/winewayland.drv/wayland.c
@@ -151,6 +151,11 @@ static void registry_handle_global(void *data, struct wl_registry *registry,
             WARN("Only a single seat is currently supported, ignoring additional seats.\n");
             return;
         }
+        if (version < 3)
+        {
+            ERR("wl_seat version 3 not supported, Aborting!\n");
+            return;
+        }
         pthread_mutex_lock(&seat->mutex);
         seat->wl_seat = wl_registry_bind(registry, id, &wl_seat_interface,
                                          version < 5 ? version : 5);
-- 
2.56.0

